doc/nrf/libraries/networking/rest_client.rst

```diff
@@ -0,0 +1,44 @@
+.. _lib_rest_client:
+
+REST client
+###########
+
+.. contents::
+   :local:
+   :depth: 2
+
+The REST client library provides a way to perform REST request.
+
+Overview
+********
+
+REST client library creates a socket with TLS if requested and uses Zephyr's HTTP client library to send HTTP request and receive HTTP response.
+The library is used by populating :c:struct:`rest_client_req_resp_context` structure passed to :c:func:`rest_client_request` function
+and the same structure will contain the response data.
+
+Configuration
+*************
+
+To enable the REST client library, enable the :kconfig:`CONFIG_REST_CLIENT` Kconfig option.
+
+Following are the options that can be configured to alter the behavior of the library:
+
+*  :kconfig:`CONFIG_REST_CLIENT_REQUEST_TIMEOUT`
+*  :kconfig:`CONFIG_REST_CLIENT_SCKT_SEND_TIMEOUT`
+*  :kconfig:`CONFIG_REST_CLIENT_SCKT_RECV_TIMEOUT`
+*  :kconfig:`CONFIG_REST_CLIENT_SCKT_TLS_SESSION_CACHE_IN_USE`
+
+Limitations
+***********
+
+*  Executing REST request is a blocking operation.
```

Comment:
This is a bit unclear, at least to me. Can you explain a bit more? And if this is the only limitation, you don't need the bullet point.